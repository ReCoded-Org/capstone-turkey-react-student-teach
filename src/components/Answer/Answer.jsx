import { useState } from 'react';
import PropTypes from 'prop-types';
import { FaEllipsisV, FaThumbsDown, FaThumbsUp } from 'react-icons/fa';

function Answer({ children, image, username }) {
  const [open, setOpen] = useState(false);
  return (
    <div className="flex flex-col mt-8 max-w-[50rem]">
      <div className="group relative bg-gray-100 rounded-lg px-5 py-4">
        <div className="flex justify-between items-center mb-2">
          <div className="flex items-center">
            <a href="/user-profile">
              <img
                src={image}
                alt={`${username}'s profile pic`}
                className="w-9 rounded-full mr-2"
              />
            </a>
            <a href="/user-profie">
              <h3 className="font-medium">{username}</h3>
            </a>
          </div>
          <button
            className=" opacity-0 group-hover:opacity-100 text-gray-500 hover:text-gray-600 hover:bg-gray-200 transition rounded-full p-2"
            data-dropdown-toggle="dropdown"
            id="dropdown"
            onClick={() => setOpen(!open)}
            type="button"
          >
            <FaEllipsisV />
          </button>
          <div
            className={`absolute top-16 right-5 z-10 w-44 border rounded shadow-lg bg-white transition py-1 ${
              open ? 'show opacity-100' : 'hidden opacity-0'
            }`}
            id="dropdown"
          >
            <button
              className="w-full text-left text-gray-800 hover:text-gray-900 hover:bg-gray-200 transition px-4 py-1"
              type="button"
            >
              Edit answer
            </button>
            <button
              className="w-full text-left text-red-600 hover:text-red-700 hover:bg-gray-200 transition px-4 py-1"
              type="button"
            >
              Delete answer
            </button>
            <button
              className="w-full text-left text-green-700 hover:text-green-800 hover:bg-gray-200 transition px-4 py-1"
              type="button"
            >
              Mark as correct answer
            </button>
          </div>
        </div>
        {children}
      </div>
      <div className="flex justify-end px-5 py-3">
        <BelowButton className="text-gray-500 hover:text-gray-600">
          <FaThumbsUp />
        </BelowButton>
        <BelowButton className="text-gray-500 hover:text-gray-600">
          <FaThumbsDown />
        </BelowButton>
        <BelowButton className="text-sm text-orange-600 hover:text-orange-700">
          Reply
        </BelowButton>
      </div>
    </div>
  );
}
Answer.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  image: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
};

function BelowButton({ children, className }) {
  return (
    <button
      type="button"
      className={`w-16 flex justify-center items-center bg-gray-100 hover:bg-gray-200 transition rounded-full py-1 mx-1 ${className}`}
    >
      {children}
    </button>
  );
}
BelowButton.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  className: PropTypes.string,
};
BelowButton.defaultProps = {
  children: '',
  className: '',
};

export default Answer;
