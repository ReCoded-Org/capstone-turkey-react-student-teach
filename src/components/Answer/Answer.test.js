import renderer from 'react-test-renderer';

import Answer from './Answer';

it('renders an answer component', () => {
  const tree = renderer
    .create(
      <Answer username="suateneskoc" image="https://via.placeholder.com/150">
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry. Lorem Ipsum has been the industry standard dummy text ever
          since the 1500s, when an unknown printer took a galley of type and
          scrambled it to make a type specimen book. It has survived not only
          five centuries, but also the leap into electronic typesetting,
          remaining essentially unchanged. It was popularised in the 1960s with
          the release of Letraset sheets containing Lorem Ipsum passages, and
          more recently with desktop publishing software like Aldus PageMaker
          including versions of Lorem Ipsum.
        </p>
      </Answer>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
