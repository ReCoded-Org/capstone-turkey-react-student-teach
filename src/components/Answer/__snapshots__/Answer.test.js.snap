// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an answer component 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="group bg-gray-100 rounded-lg px-5 py-4"
  >
    <div
      className="flex justify-between items-center mb-2"
    >
      <div
        className="flex items-center"
      >
        <a
          href="/user-profile"
        >
          <img
            alt="suateneskoc's profile pic"
            className="w-9 rounded-full mr-2"
            src="https://via.placeholder.com/150"
          />
        </a>
        <a
          href="/user-profie"
        >
          <h3
            className="font-medium"
          >
            suateneskoc
          </h3>
        </a>
      </div>
      <button
        className=" opacity-0 group-hover:opacity-100 text-gray-500 hover:text-gray-600 hover:bg-gray-200 transition rounded-full p-2"
        data-dropdown-toggle="dropdown"
        id="dropdown"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 192 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <div
        className="absolute top-16 right-5 z-10 w-44 border rounded shadow-lg bg-white transition py-1 hidden opacity-0"
        id="dropdown"
      >
        <button
          className="w-full text-left text-gray-800 hover:text-gray-900 hover:bg-gray-200 transition px-4 py-1"
          type="button"
        >
          Edit answer
        </button>
        <button
          className="w-full text-left text-red-600 hover:text-red-700 hover:bg-gray-200 transition px-4 py-1"
          type="button"
        >
          Delete answer
        </button>
        <button
          className="w-full text-left text-green-700 hover:text-green-800 hover:bg-gray-200 transition px-4 py-1"
          type="button"
        >
          Mark as correct answer
        </button>
      </div>
    </div>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </p>
  </div>
  <div
    className="flex justify-end px-5 py-3"
  >
    <button
      className="w-16 flex justify-center items-center bg-gray-100 hover:bg-gray-200 transition rounded-full py-1 mx-1 text-gray-500 hover:text-gray-600"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
        />
      </svg>
    </button>
    <button
      className="w-16 flex justify-center items-center bg-gray-100 hover:bg-gray-200 transition rounded-full py-1 mx-1 text-gray-500 hover:text-gray-600"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 56v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56zm40 200c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm272 256c-20.183 0-29.485-39.293-33.931-57.795-5.206-21.666-10.589-44.07-25.393-58.902-32.469-32.524-49.503-73.967-89.117-113.111a11.98 11.98 0 0 1-3.558-8.521V59.901c0-6.541 5.243-11.878 11.783-11.998 15.831-.29 36.694-9.079 52.651-16.178C256.189 17.598 295.709.017 343.995 0h2.844c42.777 0 93.363.413 113.774 29.737 8.392 12.057 10.446 27.034 6.148 44.632 16.312 17.053 25.063 48.863 16.382 74.757 17.544 23.432 19.143 56.132 9.308 79.469l.11.11c11.893 11.949 19.523 31.259 19.439 49.197-.156 30.352-26.157 58.098-59.553 58.098H350.723C358.03 364.34 384 388.132 384 430.548 384 504 336 512 312 512z"
        />
      </svg>
    </button>
    <button
      className="w-16 flex justify-center items-center bg-gray-100 hover:bg-gray-200 transition rounded-full py-1 mx-1 text-sm text-orange-600 hover:text-orange-700"
      type="button"
    >
      Reply
    </button>
  </div>
</div>
`;
